types/fabric/fabric-impl.d.ts

```diff
@@ -170,16 +170,18 @@ interface IDataURLOptions {
     withoutShadow?: boolean | undefined;
 }
 
-interface IEvent<E extends Event = Event> {
+interface IEvent<E extends Event = Event & MouseEvent> {
```

Comment:
Why is this change needed?

Reply:
e.ctrlKey 
e.shiftKey 
e.altKey 
e.metaKey

Reply:
or make on is generic, like canvas.on< MouseEvent >('', e => { /* e.e is MouseEvent */ });

Reply:
It's better to make on generic as you suggested, since not every event will be a mouse event. Thank you for the details!